data-science-onramp/data-ingestion/setup-test.py

```diff
@@ -0,0 +1,210 @@
+
+waiting_cluster_callback = False
+
+# Set global variables
+project = os.environ['GCLOUD_PROJECT']
+region = "us-central1"
+zone = "us-central1-a"
+cluster_name = 'setup-test-{}'.format(str(uuid.uuid4()))
+bucket_name = 'setup-test-code-{}'.format(str(uuid.uuid4()))
+
+
+@pytest.fixture(autouse=True)
+def teardown():
+    yield
+
+    # Delete cluster
+    cluster_client = dataproc.ClusterControllerClient(client_options={
+        'api_endpoint': f'{region}-dataproc.googleapis.com:443'
+    })
+
+    try:
+        operation = cluster_client.delete_cluster(project, region,
+                                                  cluster_name)
+        operation.result()
+    except GoogleAPICallError:
+        pass
+
+    # Delete GCS bucket
+    storage_client = storage.Client()
+    try:
+        bucket = storage_client.get_bucket(bucket_name)
+        bucket.delete(force=True)
+    except NotFound:
+        pass
+
+
+def test_setup(capsys):
+    '''Tests setup.py by submitting it to a dataproc cluster'''
+
+    # Create GCS Bucket
+    storage_client = storage.Client()
+    bucket = storage_client.create_bucket(bucket_name)
+
+    # Upload file
+    destination_blob_name = "setup.py"
+    blob = bucket.blob(destination_blob_name)
+    blob.upload_from_filename("setup.py")
+
+    job_file_name = "gs://" + bucket_name + "/setup.py"
+
+    # Create cluster configuration
+    zone_uri = \
+        'https://www.googleapis.com/compute/v1/projects/{}/zones/{}'.format(
+            project, zone)
+    cluster_data = {
+        'project_id': project,
+        'cluster_name': cluster_name,
+        'config': {
+            'gce_cluster_config': {
+                'zone_uri': zone_uri,
+                "metadata": {
+                    "PIP_PACKAGES": "google-cloud-storage"
+                },
+            },
+            'master_config': {
+                'num_instances': 1,
+                'machine_type_uri': 'n1-standard-8'
+            },
+            'worker_config': {
+                'num_instances': 6,
+                'machine_type_uri': 'n1-standard-8'
+            },
+            "initialization_actions": [
+                {
+                    "executable_file": ("gs://dataproc-initialization-actions/"
+                                        "python/pip-install.sh"),
+                }
+            ],
+            "software_config": {
+                "image_version": "1.5.4-debian10",
+                "optional_components": [
+                    "ANACONDA"
+                ],
+            }
+        }
+    }
+
+    # Create cluster using cluster client
+    cluster_client = dataproc.ClusterControllerClient(client_options={
+        'api_endpoint': '{}-dataproc.googleapis.com:443'.format(region)
+    })
+
+    cluster = cluster_client.create_cluster(project, region, cluster_data)
+    cluster.add_done_callback(callback)
+
+    # Wait for cluster to provision
+    global waiting_cluster_callback
+    waiting_cluster_callback = True
```

Comment:
The docs are slightly misleading, as you can actually just do the following to accomplish the same thing:
```
operation = cluster_client.create_cluster(project, region, cluster_data)
result = operation.result() #This is blocking and will not proceed with the rest of the code until completion.
```

Reply:
I defer to Brad on this for how to actually do it, but I'd like to see this accomplished without a `global`

Reply:
This should be ok to be removed altogether